src/main/java/uk/ac/ebi/embl/api/validation/check/file/ChromosomeListFileValidationCheck.java

```diff
@@ -36,33 +35,28 @@ public ChromosomeListFileValidationCheck(SubmissionOptions options) {
 	}
 
 	@Override
-	public boolean check(SubmissionFile submissionFile) throws ValidationEngineException
+	public ValidationResult check(SubmissionFile submissionFile) throws ValidationEngineException
 	{
-		boolean valid =true;
+		ValidationResult validationResult = new ValidationResult();
 		Origin origin =null;
 		try
 		{
 			clearReportFile(getReportFile(submissionFile));
 
 			if(!validateFileFormat(submissionFile.getFile(), uk.ac.ebi.embl.api.validation.submission.SubmissionFile.FileType.CHROMOSOME_LIST))
 			{
-				ValidationResult result = new ValidationResult();
-				valid = false;
-				result.append(FlatFileValidations.message(Severity.ERROR, "InvalidFileFormat","chromosome_list"));
-				if(getOptions().reportDir.isPresent())
-				getReporter().writeToFile(getReportFile(submissionFile), result);
-				addMessagekey(result);
-				return valid;
+				addErrorAndReport(validationResult,submissionFile, "InvalidFileFormat","chromosome_list");
+				return validationResult;
 			}
 			ChromosomeListFileReader reader = new ChromosomeListFileReader(submissionFile.getFile());
 
-			ValidationResult parseResult = reader.read();
-			if(!parseResult.isValid())
+			validationResult.append(reader.read());
+
+			if(!validationResult.isValid())
```

Comment:
This will be in one or two files (chromosome_list, unocalised_list)  where we read the file in one shot ,we do not iterate, so thought creating parseResult object is unnecessary. 

Regarding the code reusability and proper exception handling, I have a plan to do all those changes after this one i,e 1)proper exception handling 2)update SubmissionValidationPlan and other files to follow some standards to re-use code/and to make easily extensible. This pull request is mainly to throw some useful validation error message to stderr, so that we can query that from pipelite_stage table.

Reply:
ok